profile/profile.go

```diff
@@ -704,6 +704,21 @@ func (s *Sample) HasLabel(key, value string) bool {
 	return false
 }
 
+// GetSourceAndDiffBaseSamples returns source samples for diff and diff base
+// profiles, separated into two slices.
+func (p *Profile) GetSourceAndDiffBaseSamples() ([]*Sample, []*Sample) {
```

Comment:
This looks overly specialized and too low-level in terms of its return types to me to be a part of API when I look at it. I would probably just expose a string constant like DiffBaseLabel for "pprof::base" and keep the rest of the code in the caller.

Reply:
Done.